### PR TITLE
Mz clusters add introspection columns

### PR DESCRIPTION
**Pull request by jubrad**
### Motivation
We should show these as they may cause replicas to be redeployed on `alter cluster` if changed.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/adapter/src/catalog/builtin_table_updates.rs
Comment (jubrad) [inline src/adapter/src/catalog/builtin_table_updates.rs:279]: I'm not sure about this, I stole it from the cluster_schedules table...

File: src/catalog/src/builtin.rs
Comment (jkosh44) [inline src/catalog/src/builtin.rs:2504]: Let's move these columns above the `with_key` call.

File: src/adapter/src/catalog/builtin_table_updates.rs
Comment (jkosh44) [reply src/adapter/src/catalog/builtin_table_updates.rs:279]: Yeah, this is fine. We create the `Duration` from an `Interval` in planning, so converting it back should always succeed.

**Review by jkosh44 — APPROVED** [review]
LGTM, can you also update the docs: https://github.com/MaterializeInc/materialize/blob/72a363f6711ede1fb24a395041b21f6c84c90aae/doc/user/content/sql/system-catalog/mz_catalog.md?plain=1#L105-L120.

You'll also need to wait for SQL council approval before merging.

File: doc/user/content/sql/system-catalog/mz_catalog.md
Comment (benesch) [inline doc/user/content/sql/system-catalog/mz_catalog.md:121]: We should match the public `CREATE CLUSTER` wording here (`INTROSPECTION DEBUGGING`), rather than the internal wording int he code.

**Review by benesch — CHANGES_REQUESTED** [review]
One minor tweak on behalf of the SQL council. Thanks for doing this!

File: doc/user/content/sql/system-catalog/mz_catalog.md
Comment (benesch) [inline doc/user/content/sql/system-catalog/mz_catalog.md:121]:
```suggestion
| `introspection_debugging` | [`boolean`]          | Whether introspection of the gathering of the introspection data is enabled. |
```

File: doc/user/content/sql/system-catalog/mz_catalog.md
Comment (benesch) [inline doc/user/content/sql/system-catalog/mz_catalog.md:121]:
```suggestion
| `introspection_debugging` | [`boolean`]          | Whether introspection of the gathering of the introspection data is enabled.                                                             |
```

**Review by benesch — APPROVED** [review]
LGTM from a SQL council perspective. 🎉 Thanks again for knocking this one out!

**Pull request merged**
